packages/x-scheduler/src/primitives/utils/recurrence-utils.ts

```diff
@@ -478,13 +570,38 @@ export function countMonthlyOccurrencesUpToExact(
 
   const interval = Math.max(1, rule.interval ?? 1);
 
-  // TODO (#19128): Add support for monthly recurrence modes (BYDAY rules)
+  // Path A: BYDAY with ordinals (e.g. 2TU, -1FR). Not mixed with BYMONTHDAY.
   if (rule.byDay?.length) {
-    // Only "same day-of-month" mode is supported right now.
-    // If a MONTHLY rule provides BYDAY (e.g., 2TU, -1FR), we intentionally IGNORE it for now.
+    if (rule.byMonthDay?.length) {
+      throw new Error('MONTHLY: use either BYDAY (ordinal) or BYMONTHDAY, not both.');
```

Comment:
Can you prefix all the errors with `Event Calendar: `
We will discuss the exact prefix in the future, but there will be one 👍